### PR TITLE
- adding basic video demo, with no webgl for testing on mobile.

### PR DESCRIPTION
**Pull request by danrossi**
- adding basic video demo, with no webgl for testing on mobile.
- fixes for build dependancies.

### CODE REVIEW

**Comment by danrossi** [comment]
I signed it!

**Comment by danrossi** [comment]
creepy googlebot

**Comment by hoch** [comment]
Thanks for your contribution, I would suggest you to revise this PR before I can accept it.

1. This video is not a part of Omnitone library.
2. This URL can be changed anytime then this demo will not work anymore.
3. This video is not supposed to be distributed publicly. We have a special permission to use them only with the demo player.

**Comment by hoch** [comment]
What is the `-` sign in the title?

File: examples/video.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      // Set up the context and the audio element.
+      var audioContext = new AudioContext();
+      var videoElement = document.createElement('video');
+      videoElement.loop = true;
+      videoElement.autoplay = true;
+      videoElement.crossOrigin = 'anonymous';
+      videoElement.controls = true;
+      videoElement.width = 600;
+      videoElement.height = 400;
+      videoElement.src = "https://storage.googleapis.com/omnitone-demo.google.com.a.appspot.com/fuerza-imprevista-1080p-h264-aac.mp4";
```
Comment (hoch) [inline examples/video.html]: If you want to include a video, you have to use something publicly available (no copyright issue) and small enough to be included in this repository.

File: examples/video.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      //videoElement.src = 'https://storage.googleapis.com/omnitone-demo.google.com.a.appspot.com/fuerza-imprevista-1080p-vp9-vorbis.webm';
+
+      document.body.appendChild(videoElement);
+
+      // Create a decoder.
+      var decoder = Omnitone.createFOADecoder(audioContext, videoElement, {
+        // The example audio is in the FuMa ordering (W,X,Y,Z). So remap the
+        // channels to the ACN format.
+       // channelMap: [0, 3, 1, 2]
```
Comment (hoch) [inline examples/video.html]: Incorrect indentation.

File: examples/video.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      });
+
+
+      videoElement.onclick = function () {
+        videoElement.play();
+      };
+
+      // Initialize the decoder and activate the button when ready.
+      decoder.initialize().then(function () {
+        //decoder.mode = "none";
```
Comment (hoch) [inline examples/video.html]: Ditto. Explain why this line is commented out.

File: src/audiobuffer-manager.js [hunk 4]
```diff
@@ -81,7 +81,7 @@ AudioBufferManager.prototype._loadAudioFile = function (fileInfo) {
   // TODO: fetch local resources if XHR fails.
   xhr.onerror = function (event) {
     Omnitone.LOG('XHR Network failure: ' + fileInfo.url);
-    me._done(fileInfo.name, null);
+    that._done(fileInfo.name, null);
```
Comment (hoch) [inline src/audiobuffer-manager.js]: Well, this is an obvious mistake from me. :(

File: package.json [hunk 5]
```diff
@@ -11,6 +11,7 @@
     "Web Audio API"
   ],
   "devDependencies": {
+    "copy-webpack-plugin": "^3.0.1",
```
Comment (hoch) [inline package.json]: Nice catch!

**Comment by hoch** [comment]
@danrossi Could you please read my comments before pushing more fixes to this PR?

**Comment by danrossi** [comment]
@hoch I'm sorry I was not aware of that. 

I merely resized it to get testing working on mobile especially Firefox. What video can be put in there  ? 

I can try to author a Big Buck Bunny 2D from it's surround mix to ambisonics. To at least get a video tag example in there ?  That could take a while but I need to work out myself how to deliver from surround mixes anyway. 

Another step would be documenting rotation. 

I can reach out to the group on FB for a demo video ? 

Or remove the demo completely and document in the readme ?

**Comment by hoch** [comment]
> I merely resized it to get testing working on mobile especially Firefox. What video can be put in there ?

See what I have done for the audio example. If you can find an equivalent of that, I think it is safe to be a part of the repository.

https://github.com/GoogleChrome/omnitone/tree/master/examples/resources

> Another step would be documenting rotation.

What does this mean?

> Or remove the demo completely and document in the readme ?

I think it's better to remove the demo from this PR, so I can merge your fix asap. You can always try again when you have an appropriate video example.

**Comment by danrossi** [comment]
Closing and will try again.

**Pull request closed without merging**